lib/Access.php

```diff
@@ -606,25 +606,39 @@ public function dn2ocname($fdn, $ldapDisplayName = null, $isUser = true) {
 		// outside of core user management will still cache the user as non-existing.
 		$originalTTL = $this->connection->ldapCacheTTL;
 		$this->connection->setConfiguration(['ldapCacheTTL' => 0]);
-		// FIXME DI User and Group Managers
-		if (($isUser && !\OC::$server->getUserManager()->userExists($intName))
+		if (($isUser && $this->shouldMapToUsername($intName))
 			|| (!$isUser && !\OC::$server->getGroupManager()->groupExists($intName))) {
+			\OC::$server->getLogger()->info("Reusing existing mapping for ownCloud UUID: $intName to LDAP UUID: $uuid", ['app' => 'user_ldap']);
 			if ($mapper->map($fdn, $intName, $uuid)) {
 				$this->connection->setConfiguration(['ldapCacheTTL' => $originalTTL]);
 				return $intName;
 			}
 		}
 		$this->connection->setConfiguration(['ldapCacheTTL' => $originalTTL]);
+		\OC::$server->getLogger()->error("Mapping collision for DN $fdn and UUID $uuid. Couldn't map to: $intName", ['app' => 'user_ldap']);
+		//if everything else did not help..
+		return false;
```

Comment:
`false` might not be the correct response. `dn2ocname` will be called by several methods in the core `UserManager` and the `Groups`. If something dies we do log an error ... so the admin should be able to find the root cause. That is if the calling method knows how to deal with `false`. Not nice but good enough because the old code also used `false`. :man_shrugging: